charms/istio-pilot/src/charm.py

```diff
@@ -828,7 +894,97 @@ def _istiod_svc(self):
         else:
             return exporter_ip
 
-    def _use_https(self):
+    @property
+    def _ssl_info(self) -> Dict[str, str]:
+        """Return a dictionary with SSL cert and key values.
+
+        The dictionary is built based on available information, if
+        the istio-tls-secret is found, it is prioritised and returned;
+        otherwise, the information shared by a TLS certificate provider.
+        """
+
+        # FIXME: remove the try/catch block and just return the dictionary that contains
+        # data from the CertHandler after 1.18
+        try:
+            ssl_secret = self.model.get_secret(label="istio-tls-secret")
+            return {
+                "ssl-crt": base64.b64encode(
+                    ssl_secret.get_content()["ssl-crt"].encode("ascii")
+                ).decode("utf-8"),
+                "ssl-key": base64.b64encode(
+                    ssl_secret.get_content()["ssl-key"].encode("ascii")
+                ).decode("utf-8"),
+            }
+        except SecretNotFoundError:
+            return {
+                "ssl-crt": base64.b64encode(self._cert_handler.cert.encode("ascii")).decode(
+                    "utf-8"
+                ),
+                "ssl-key": base64.b64encode(self._cert_handler.key.encode("ascii")).decode(
+                    "utf-8"
+                ),
+            }
+
+    # ---- Start of the block
+    # ---- WARNING: this feature is not recommended, but is supported in 1.17 and 1.18.
+    # ---- For details please refer to canonical/istio-operators#380.
+    # ---- FIXME: Remove this block after releasing 1.18.
+    def _use_https(self) -> bool:
+        """Return True if only one of the TLS configurations are enabled, False if none are.
+
+        Raises:
+            ErrorWithStatus: if both configurations are enabled.
+        """
+        if self._use_https_with_tls_provider() and self._use_https_with_tls_secret():
+            self.log.info(
+                "Only one TLS configuration is supported at a time."
+                "Either remove the TLS certificate provider relation,"
+                "or the TLS manual configuration with the remove-tls-secret action."
+            )
```

Comment:
I think this should be a `warn` (or an `error`?) since we're blocking for this reason